### PR TITLE
feat: Add support for deleting threads

### PR DESCRIPTION
**Pull request by FlowFlorent**
### Description

**@liveblocks/react**

Room & thread cache now handle thread deletion by listening to THREAD_DELETED event

**@liveblocks/node**
- Adds `threadDeleted` webhook event
- Adds `deleteThread` method to client

### CODE REVIEW

File: packages/liveblocks-core/src/protocol/ServerMsg.ts [hunk 1]
```diff
@@ -26,6 +26,7 @@ export enum ServerMsgCode {
   COMMENT_DELETED = 404,
   COMMENT_REACTION_ADDED = 405,
   COMMENT_REACTION_REMOVED = 406,
+  THREAD_DELETED = 407,
```
Comment (nvie) [inline packages/liveblocks-core/src/protocol/ServerMsg.ts:29]: We can safely move this a couple lines higher, together with the other `THREAD_` enum members. You don't have to worry about not noticing the code if we add new members later. We already have a lint rule that will fail if there are duplicate members.

**Review by nvie — APPROVED** [review]
💯

File: packages/liveblocks-core/src/protocol/ServerMsg.ts (hunk 1 above)
Comment (FlowFlorent) [reply packages/liveblocks-core/src/protocol/ServerMsg.ts:29]: nice, adjusting :)

File: packages/liveblocks-node/src/client.ts [hunk 2]
```diff
@@ -1224,6 +1224,25 @@ export class Liveblocks {
...
+   * @param params.threadId The thread ID to delete.
+   */
+  public async deleteThread(params: {
+    roomId: string;
+    threadId: string;
+  }): Promise<void> {
+    const { roomId, threadId } = params;
+
+    const res = await this.delete(url`/v2/rooms/${roomId}/threads/${threadId}`);
+    console.log("deleteThread res", res);
```
Comment (FlowFlorent) [inline packages/liveblocks-node/src/client.ts:1239]:
```suggestion
```

**Pull request merged**